sdk/typescript/lib/subapis/transaction.ts

```diff
@@ -0,0 +1,31 @@
+import { TransactionApi } from '../generated'
+
+export class Transaction {
+  constructor(public innerClient: TransactionApi) {}
+
+  getStatus(transactionIntentHashHex: string) {
+    return this.innerClient.transactionStatus({
+      transactionStatusRequest: {
+        intent_hash_hex: transactionIntentHashHex,
+      },
+    })
+  }
+
+  getCommitedDetails(
+    transactionIntentHashHex: string,
+    stateVersion?: number
```

Comment:
Why would you imagine they'd use the state version?

Relatedly - could we add JS Docs to all the methods explaining how they should be used / what they do?

Reply:
Dashboard uses it in order to display ledger state at the moment of submitting TX. We first query by TxID, then get state version, then query gateway again with that state version in order to get `epoch` and `round` at the time TX was submitted.

Now when you're asking I'm thinking I maybe should wrap those two calls inside SDK?

Reply:
It sounds like either we should change committed details to return those details, or yes, we create some helper which does these two queries for you  - or maybe separate the queries?

I'm surprised we don't return the ledger state identifiers (eg epoch/round/state version) of a committed transaction already? Maybe this would be an easy request to the gateway team if it's not there.

Reply:
Gateway does return **current** ledger state whereas what we want is ledger state at the moment of tx submission. I guess I could use another endpoint that basically returns ledger state (epoch&round) for given state version.